drivers/FileBase.cpp

```diff
@@ -52,8 +53,8 @@ FileBase::~FileBase() {
     _mutex->unlock();
 
     if (getPathType() == FilePathType) {
-        extern void remove_filehandle(FileLike *file);
-        remove_filehandle(static_cast<FileLike*>(this));
+        extern void remove_filehandle(FileHandle *file);
```

Comment:
Hmm, that's an interesting question. It sounds like poll should go in its own file? There are a few other non-member functions like sleep and wait that get their own files in a similar manner.